src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -1282,4 +1282,47 @@ BOOST_AUTO_TEST_CASE(SimpleMeanSquaredErrorLayerTest)
   BOOST_REQUIRE_EQUAL(output.n_elem, 1);
 }
 
+/*
+ * Simple test for the BilinearInterpolation layer
+ */
+BOOST_AUTO_TEST_CASE(SimpleBilinearInterpolationLayerTest)
```

Comment:
Another idea would be to compare against armadillo's interpolation method on an 1D vector.

Reply:
I'll add a small example soon.

Reply:
Great, thanks!

Reply:
@zoq I checked for `arma::interp1()`, and it seems that it doesn't support matrix types. So basically, what I'll have to do is create a number of row vectors and first interpolate them along one direction (along row), and then create another set of vectors from the column entries of these row vectors and interpolate them along the columns. Only then we'd be able to compare the methods. This would be tedious for a good test (2x2 to 3x3 matrix is doable, but any higher would just be time consuming). How big do we want the test to be?

Reply:
I would say that 2x2 or 3x3 is just fine.